quesma/queryparser/pancake_transformer.go

```diff
@@ -408,48 +408,61 @@ func (a *pancakeTransformer) aggregationTreeToPancakes(topLevel pancakeAggregati
 		// TODO: if both top_hits/top_metrics, and filters, it probably won't work...
 		// Care: order of these two functions is unfortunately important.
 		// Should be fixed after this TODO
-		newFiltersPancakes := a.createFiltersPancakes(&newPancake)
+		newCombinatorPancakes := a.createCombinatorPancakes(&newPancake)
 		additionalTopHitPancakes, err := a.createTopHitAndTopMetricsPancakes(&newPancake)
 		if err != nil {
 			return nil, err
 		}
 
 		pancakeResults = append(pancakeResults, additionalTopHitPancakes...)
-		pancakeResults = append(pancakeResults, newFiltersPancakes...)
+		pancakeResults = append(pancakeResults, newCombinatorPancakes...)
 	}
 
 	return
 }
 
 // createFiltersPancakes only does something, if first layer aggregation is Filters.
 // It creates new pancakes for each filter in that aggregation, and updates `pancake` to have only first filter.
-func (a *pancakeTransformer) createFiltersPancakes(pancake *pancakeModel) (newPancakes []*pancakeModel) {
+func (a *pancakeTransformer) createCombinatorPancakes(pancake *pancakeModel) (newPancakes []*pancakeModel) {
 	if len(pancake.layers) == 0 || pancake.layers[0].nextBucketAggregation == nil {
 		return
 	}
 
 	firstLayer := pancake.layers[0]
-	filters, isFilters := firstLayer.nextBucketAggregation.queryType.(bucket_aggregations.Filters)
-	canSimplyAddFilterToWhereClause := len(firstLayer.currentMetricAggregations) == 0 && len(firstLayer.currentPipelineAggregations) == 0
-	areNewPancakesReallyNeeded := len(pancake.layers) > 1 // if there is only one layer, it's better to get it done with combinators.
+	combinator, isCombinator := firstLayer.nextBucketAggregation.queryType.(bucket_aggregations.CombinatorAggregationInterface)
+	if !isCombinator {
+		return
+	}
+
+	noMoreBucket := len(pancake.layers) <= 1 || (len(pancake.layers) == 2 && pancake.layers[1].nextBucketAggregation == nil)
+	noMetricOnFirstLayer := len(firstLayer.currentMetricAggregations) == 0 && len(firstLayer.currentPipelineAggregations) == 0
+	canSimplyAddCombinatorToWhereClause := noMoreBucket && noMetricOnFirstLayer
+	if canSimplyAddCombinatorToWhereClause {
+		return
+	}
 
```

Comment:
I think code should be a bit clearer now? Feel free to disagree

Reply:
Though not sure if that's always correct if we count size. Though it might be.

Reply:
Ok, we add size as metric aggregation, so this should be good.